pulp_rpm/app/tasks/publishing.py

```diff
@@ -257,6 +267,14 @@ def create_repomd_xml(content, publication, extra_repomdrecords,
     # Process all packages
     for package in packages.iterator():
         pkg = package.to_createrepo_c()
+        pkg_filename = os.path.basename(package.location_href)
+        # this can cause an issue when two same RPM package names appears
+        # a/name1.rpm b/name1.rpm
```

Comment:
i do not think this is an issue yet, since we have not added location_href among repo_key fields.

Reply:
or let' say - this is an issue that would not be managed during sync, so sync would fail if we would have 2 same rpms what would differ only in the location_href.

Reply:
so i did some more testing where:
- uploaded artifact A
- created an rpm out out it  and used relative_path X
- created an rpm out of artifact A and used relative_path Y - this task failed with an error
```
        "description": "{'non_field_errors': [ErrorDetail(string='There is already a package with: arch=noarch, checksum_type=sha256, epoch=0, name=fox, pkgId=eff95a379774c7622037bcb7ff347271dc727ae4ddce9b16017477b58488aa3e, release=2, version=1.1.', code='invalid')]}",
```

Then i tried:
- uploaded artifact A
- created an rpm out out it  and used relative_path X
- uploaded artifact B
- created an rpm out out it  and used relative_path X.
- 2 rpms got uploaded into pulp, however then i tried to add those 2 rpms to the repo , I got an error
```
"description": "Cannot create repository version. Path is duplicated: fox-1.1-2.noarch.rpm.",
```

This testing has shown that:
a) within a repo version there cannot be 2 rpms with the same relative_path  [aka our location_href from rpm during sync `os.path.basename(package.location_href)` or the relative_path we provide during upload and eventually treat it as location_href]
b) within pulp we cannot have 2 same rpms with different relative_path